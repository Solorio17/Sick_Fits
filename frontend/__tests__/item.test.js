import Item from '../components/Item';
import { shallow, mount } from 'enzyme';
import toJSON from 'enzyme-to-json';

const fakeItem = {
    id: '34567o8fuvkyudgw5t3',
    title: 'A very neat item',
    price: 7000,
    description: 'This item is very cool and handy',
    image: 'item.jpg',
    largeImage: 'itemLarge.jpg'
}

describe('Item component', () => {
    //SAME THING AS BELOW BUT WITH A SNAPSHOT
    it('renders and matches the snapshot', ()=>{
        const wrapper = shallow(<Item item={fakeItem}/>)
        expect(toJSON(wrapper)).toMatchSnapshot()
    })


    // it('renders and displays properly', ()=>{
    //     const wrapper = shallow(<Item item={fakeItem}/>);
    //     const PriceTag = wrapper.find('PriceTag');
    //     expect(PriceTag.dive().text()).toBe('$50');

    //     expect(wrapper.find('Title a').text()).toBe(fakeItem.title);

    //     const img = wrapper.find('img');
    //     expect(img.props().src).toBe(fakeItem.image);
    //     expect(img.props().alt).toBe(fakeItem.title);

    //     const buttonList = wrapper.find('.buttonList');
    //     expect(buttonList.children()).toHaveLength(3);
    //     expect(buttonList.find('Link')).toHaveLength(1);
    //     expect(buttonList.find('AddToCart').exists()).toBe(true);
    //     expect(buttonList.find('RemoveFromCart')).toBeTruthy();
    // })
})