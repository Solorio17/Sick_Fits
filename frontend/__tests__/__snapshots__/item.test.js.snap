// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item component renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "34567o8fuvkyudgw5t3",
          },
        }
      }
    >
      <a>
        A very neat item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $70
  </PriceTag>
  <img
    alt="A very neat item"
    src="item.jpg"
  />
  <p>
    This item is very cool and handy
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "34567o8fuvkyudgw5t3",
          },
        }
      }
    >
      <a>
        Edit
      </a>
    </Link>
    <AddToCart
      id="34567o8fuvkyudgw5t3"
    />
    <DeleteItem
      id="34567o8fuvkyudgw5t3"
    >
      Delete Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
